plugins/org.yakindu.sct.model.stext/src/org/yakindu/sct/model/stext/validation/STextJavaValidator.java

```diff
@@ -899,34 +905,51 @@ public void checkImportExists(ImportScope scope) {
 				error(String.format(IMPORT_NOT_RESOLVED_MSG, packageImport), scope,
 						StextPackage.Literals.IMPORT_SCOPE__IMPORTS, imports.indexOf(packageImport));
 			}
+			else {
+				IResourceDescription iResourceDescription = resourceDescriptionCache.get(pkImport.get().getUri());
+				if (iResourceDescription instanceof DefaultResourceDescription) {
+					Resource resource = ((DefaultResourceDescription) iResourceDescription).getResource();
+					if (!(resource.getWarnings().isEmpty() && resource.getErrors().isEmpty())) {
```

Comment:
Does it mean, I will now get an error in the statechart although the imported resource has just a warning?
I would prefer to not be too strict here. Actually I think an info message would be enough here.